### ISSUE TITLE
protagonist@1.5.2 installation fails on Ubuntu 16.04

### ISSUE THREAD

**Issue by Ravichandrane** [apiaryio/protagonist#169]
Hi,

Having some trouble installing protagonist on Ubuntu 16.04 with node 6.8.1 & npm 3.10.8.

Cheers,
Ravi

![screen shot 2016-10-17 at 00 22 50](https://cloud.githubusercontent.com/assets/4774729/19421396/5c865988-9400-11e6-8200-8b712edb4c0a.png)

**Comment by w-vi** [comment]
@Ravichandrane can you please post also either the `npm-debug.log` or the output before this screenshot as from the picture we have no idea what went wrong.

**Comment by Ravichandrane** [comment]
Hi @w-vi,

Here is the [npm-debug.log ](https://cl.ly/3G1n0q1j302x)

**Comment by w-vi** [comment]
@Ravichandrane So that didn't really say much, can you please show the output of  `gcc --version` and ideally capture what gets printed to console during the `npm i protagonist` to see if there are any compilation errors.

**Comment by o5** [comment]
I had similar problems with installation on Ubuntu, because `python3` is default in Ubuntu 16.04 and protagonist needs `python2` (next dependencies are `make` and `g++`). I know, there is a [notice](https://github.com/apiaryio/protagonist/commit/d9a063f80738894416cafe5044197069075a1a88) about it, but it would be great to have dockerized version of Protagonist. 

PS: I'm happy user of dockerized `apiaryio/apiary-client` :)

**Comment by kylef** [comment]
There should be a warning emitted if that was the case: https://github.com/apiaryio/drafter/blob/846065000ba390935e23fa4d0a7830f4790b6db4/configure#L12-L17 (this should be printed to stderr instead of stdout).

@w-vi If this is problematic, we could update Drafter's configure script to re-invoke itself with `python2` when `python` is not version 2. Something along the lines of:

```python
if sys.version_info[0] != 2:
    # Check if `python2` is found on path, exit with error if not

    # switch to python2 with same argv
    exec('python2', *sys.argv)
```

**Comment by w-vi** [comment]
@kylef  If python2 is now by default on Ubuntu we should really have an automatic way as you described

**Comment by EvAlex** [comment]
It fails also on Windows and in `node:latest` docker too

**Comment by Ravichandrane** [comment]
Hey guys, 

I fixed the issue on Ubuntu 16.04 with 

- `sudo apt-get install build-essential` 
- `sudo apt-get install build-essential g++`
- `sudo apt-get install build-essential make`
- `sudo npm install -g node-gyp`

I hope this can help you.

**Comment by o5** [comment]
From [comment](https://github.com/apiaryio/protagonist/issues/169#issuecomment-256036380):
> I needed something like protagonist-cli because my use case is Get JSON Schema from Blueprint

Here is what I looked for: [o5/apibjson](https://github.com/o5/apib2json) :)
